selinux/cockpit.fc

```diff
@@ -12,7 +12,7 @@
 
 /var/lib/cockpit(/.*)?      gen_context(system_u:object_r:cockpit_var_lib_t,s0)
 
-/var/run/cockpit(/.*)?   gen_context(system_u:object_r:cockpit_var_run_t,s0)
-/var/run/cockpit-ws(/.*)?   gen_context(system_u:object_r:cockpit_var_run_t,s0)
+/run/cockpit(/.*)?   gen_context(system_u:object_r:cockpit_var_run_t,s0)
+/run/cockpit-ws(/.*)?   gen_context(system_u:object_r:cockpit_var_run_t,s0)
```

Comment:
This doesn't work in F38/F39, see the failed tests. Is it possible to specify both? Like `{,/var}/run...` or just two lines?

If this doesn't work, we'll have to patch this dynamically in selinux/Makefile.am. What would a good feature test be? Or do we need to check the installed selinux-policy for version ≥ 40.12-1?

Reply:
It is not expected to work with version < selinux-policy-40.12-1.fc40, i. e. also not in F39 and older. I am not well aware how you handle packages for different releases, I can think of alternative options like to use entries for both /run and /var/run, or if statements in the policy sources.

I think it is sufficient to check that the services start. From SELinux POV it is needed that files in /run have the private type like cockpit_var_run_t, not the generic var_run_t or even a different one.

Reply:
We have a single upstream release tarball, including .spec file, which works across all supported releases (RHEL 8/debian stable all the way to Fedora rawhide/arch/etc.)

As I said we have the option of live-patching the .fc file in our Makefile rule if necessary, i.e. if specifying both doesn't work.

Reply:
Entries for both paths will work, the other is just overriden by equivalency rule.

Reply:
Thanks! Adjusted, see below.